activesupport/test/cache/behaviors/cache_store_behavior.rb

```diff
@@ -566,6 +566,28 @@ def test_expires_in_and_expires_at
     assert_equal "Either :expires_in or :expires_at can be supplied, but not both", error.message
   end
 
+  def test_invalid_expiration_time_raises_an_error_when_raise_on_invalid_cache_expiration_time_is_true
+    ActiveSupport::Cache::Store.raise_on_invalid_cache_expiration_time = true
+    key = SecureRandom.uuid
+    error = assert_raises(ArgumentError) do
+      @cache.write(key, "bar", expires_at: 1.minute.ago)
+    end
+    assert_equal "Cache expiration time is invalid", error.message
+  end
+
+  def test_invalid_expiration_time_reports_and_logs_when_raise_on_invalid_cache_expiration_time_is_false
+    ActiveSupport::Cache::Store.raise_on_invalid_cache_expiration_time = false
+    ActiveSupport.error_reporter = ActiveSupport::ErrorReporter.new
+    ActiveSupport::Cache::Store.logger = ActiveSupport::Logger.new(nil)
+    key = SecureRandom.uuid
+    error = ArgumentError.new("Cache expiration time is invalid")
+    assert_called(ActiveSupport.error_reporter, :report, [error, handled: true]) do
+      assert_called(ActiveSupport::Cache::Store.logger, :error, ["#{error}: #{error.message}"]) do
```

Comment:
I'm not a fan of using `assert_called` for this. We should have a block based test helper that adds a new error error reporter that collects the errors and either return them to you or makes some assertions on them.

e.g. 
```ruby
assert_reports_error(error, handled: true) do
  # ...
end
```